domain/relation/state/relation.go

```diff
@@ -1284,6 +1284,85 @@ ON CONFLICT (relation_uuid, unit_uuid) DO UPDATE SET
 	return tx.Query(ctx, insertStmt, insertRelationUnit).Run()
 }
 
+// LeaveScope updates the given relation to indicate it is not in scope.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.RelationUnitNotFound] if the relation unit cannot be
+//     found.
+func (st *State) LeaveScope(ctx context.Context, relationUnitUUID corerelation.UnitUUID) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Check the relation unit exists.
+		exists, err := checkExistsByUUID(ctx, st, tx, "relation_unit", relationUnitUUID.String())
```

Comment:
I'd probably log the occurrence of zero affected rows rather having an extra query for every scope departure, but we can live with this for now.